polkadot/node/primitives/src/approval.rs

```diff
@@ -532,4 +532,35 @@ mod test {
 		assert_eq!(bitfield.inner_mut().count_ones(), 1);
 		assert_eq!(bitfield.len(), 21);
 	}
+
+	#[test]
+	fn test_kagome_vrf() {
+		use parity_scale_codec::Decode as _;
+
+		let data = "0x0301000000cffde21000000000ee5d1e3c744c86836bd2b74f63b1c1ebbfcbb4b4991620ec5b768b9fcaec1b5c7b57ba0d7dbaf0c47964c49452ddff29f79356be4f9914646cf943b2a7764f02c0664d7ad081ee50500844c1fe783cdeae4ea829f888f7add890247f8ac8ba04";
+		let data = hex::decode(&data[2..]).unwrap();
+		let pre_digest = sp_consensus_babe::digests::PreDigest::decode(&mut &data[..]).unwrap();
+		let sig = pre_digest.vrf_signature().unwrap();
+		assert_eq!(pre_digest.authority_index(), 1);
+
+		// let randomness = "0x950be62c56a9b1d6487608777fe7e3966f67f5449e65102ec5174847fcfb9013";
+		let randomness = "0x4cd06ed03c0d2c818d100bbcca1b10344a229e2e22dc5c9180d99297a956500e";
+		let randomness = hex::decode(&randomness[2..]).unwrap();
+		let randomness = randomness.try_into().unwrap();
+
+		let slot = pre_digest.slot();
+
+		let author = "0x6a91dff8d85ae634ad2ff7da25e2bf15045168aa4ec230abf052bee55f683f13";
+		let author = hex::decode(&author[2..]).unwrap();
+
+		let epoch_index = 31_553;
+
+		let pubkey = schnorrkel::PublicKey::from_bytes(author.as_slice()).unwrap();
+
+		let transcript = sp_consensus_babe::make_vrf_transcript(&randomness, slot, epoch_index);
+
+		let inout = sig.output.0.attach_input_hash(&pubkey, transcript.0).unwrap();
```

Comment:
Could you try instead:

```rust
pubkey.vrf_verify(&transcript, &sig)
```

It's what BABE [uses](https://github.com/paritytech/polkadot-sdk/tree/39cc95740aa03f53a0e8bfb8537f0ddb5fa3e19e/substrate/client/consensus/babe/src/verification.rs) (`vrf_verify` will internally do the call to `attach_input_hash`).